karabo_data/geometry.py

```diff
@@ -231,11 +231,96 @@ def inspect(self):
         ax.set_title('LPD detector geometry')
         return fig
 
+    def to_distortion_array(self):
+        """ Return distortion matrix for LPD detector
+
+        Returns
+        -------
+        out: ndarray
+            Dimension (4096=16(modules)*256(ss_dim), 256(fs_dim), 4, 3)
+            type: float32
+            4 is the number of corners of a pixel
+            last dimension is for Z, Y, X location of each corner
+        """
+        _tiles = self._create_qmt_dict()
+        _, centre = self._plotting_dimensions()
+        distortion = np.zeros((4096, 256, 4, 3), dtype=np.float32)
+
+        for index, (Q, M) in enumerate(product(range(1, 5), range(1, 5))):
+            #ch: module offset along first dimension of distortion array
+            ch = index*256
+            for T in range(1, 17):
+                position = self.find_offset(('Q%d' % Q,
+                                             'M%d' % M, 'T%02d' % T))
+
+                # idx, idy : Bottom left index of tiles in (256,256)
+                #            module array
+                idx, idy = _tiles['T%01d' % T]
+
+                xt, yt = (position // self.pixel_size) + centre
+                xt, yt = int(xt), int(yt)
+                # y_position contains y_indices of pixels for a tile
+                y_position = np.array(
+                    [yt+j for j, _ in product(range(32), range(128))]).\
+                    reshape(32, 128)
```

Comment:
I think [np.meshgrid](https://docs.scipy.org/doc/numpy/reference/generated/numpy.meshgrid.html) might be useful for these.

Reply:
done.